zarr/sync.py

```diff
@@ -5,7 +5,13 @@
 import fasteners
 
 
-class ThreadSynchronizer:
+class Synchronizer:
```

Comment:
over in https://github.com/zarr-developers/zarr-python/pull/1495/files#diff-9b925c58923bb1c258699fe5c0f20a9a8cff470568d6badb9498156dd19a44e9 I use protocols here, in case that's of interest. I'm not sure when / if my PR will get merged (I haven't been pushing it at all), and I'm happy to rebase on top of your PR if yours gets merged first.

Reply:
Ah, thanks for reminding me of Protocols, I added a simple one for `Synchronizer` for now, but I imagine that will be made redundant by your PR anyway.